jbpm-console-ng-workbench-integration/jbpm-console-ng-workbench-integration-client/src/test/java/org/jbpm/console/ng/wi/client/editors/deployment/descriptor/DeploymentDescriptorEditorPresenterTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.console.ng.wi.client.editors.deployment.descriptor;
+
+import java.util.List;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.guvnor.common.services.shared.metadata.model.Metadata;
+import org.guvnor.common.services.shared.validation.model.ValidationMessage;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.ErrorCallback;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.jbpm.console.ng.wi.dd.model.DeploymentDescriptorModel;
+import org.jbpm.console.ng.wi.dd.service.DDEditorService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.internal.runtime.conf.RuntimeStrategy;
+import org.kie.workbench.common.widgets.metadata.client.KieEditorWrapperView;
+import org.kie.workbench.common.widgets.metadata.client.widget.OverviewWidgetPresenter;
+import org.mockito.Mock;
+import org.uberfire.backend.vfs.ObservablePath;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.editor.commons.client.history.VersionRecordManager;
+import org.uberfire.mvp.PlaceRequest;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class DeploymentDescriptorEditorPresenterTest {
+
+    @Mock
+    ObservablePath observablePath;
+
+    @Mock
+    DeploymentDescriptorViewImpl view;
+
+    @Mock
+    TranslationService translationService;
+
+    DeploymentDescriptorEditorPresenter presenter;
+
+    @Before
+    public void setUp() throws Exception {
+        presenter = new DeploymentDescriptorEditorPresenter(view,
+                new ServiceMock(), translationService
+        ) {
+            {
+                kieView = mock(KieEditorWrapperView.class);
+                overviewWidget = mock(OverviewWidgetPresenter.class);
+                versionRecordManager = mock(VersionRecordManager.class);
+                concurrentUpdateSessionInfo = null;
+            }
+
+            protected void makeMenuBar() {
+
+            }
+
+            protected void addSourcePage() {
+
+            }
+        };
+    }
+
+    @Test
+    public void testDeploymentDescriptorEditorSetup() throws Exception {
+
+        presenter.onStartup( mock(ObservablePath.class), mock(PlaceRequest.class) );
```

Comment:
Consider creating an init method in the presenter using `@PostConstruct`, that way your test wold be much simpler.